lib/analytics/course_revisions_csv_builder.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CourseRevisionsCsvBuilder
+  def initialize(course)
+    @course = course
+    set_revisions
+  end
+
+  def generate_csv
+    csv_data = [CSV_HEADERS]
+    revisions_rows.each do |row|
+      csv_data << row
+    end
+    CSV.generate { |csv| csv_data.each { |line| csv << line } }
+  end
+
+  def revisions_rows
+    @new_revisions.values.map do |revision_data|
+      build_row(revision_data)
+    end
+  end
+
+  def set_revisions
+    @new_revisions = {}
+    @course.all_revisions.includes(article: :wiki).map do |edit|
```

Comment:
This should probably be `includes(:wiki, :article)`. The output should include the title and namespace of the article, as well as the domain (rather than the Dashboard-specific ID) of the wiki.